server/gokbg3/grails-app/controllers/org/gokb/CreateController.groovy

```diff
@@ -224,6 +224,20 @@ class CreateController {
                 }
               }
 
+              if (result.newobj.respondsTo("getCuratoryGroups")) {
+                log.debug("Set CuratoryGroups..");
+                if(user.isAdmin() || user.getSuperUserStatus()) {
+                  flash.message = "Object was not assigned to a curator group because you are admin or superuser!!!!"
+
+                }else {
+                  user.curatoryGroups.each { CuratoryGroup cg ->
+                    result.newobj.curatoryGroups.add(cg)
+                  }
+                }
+
+                  result.newobj.save(flush:true)
```

Comment:
try to avoid flush:true whenever possible

Reply:
arranged in personal conversation - flushes will be dealt in separate refactoring